src/modal/PeraWalletConnectModal.tsx

```diff
@@ -1,131 +1,111 @@
 import CloseIcon from "../asset/icon/Close.svg";
+import CloseIconDark from "../asset/icon/Close--dark.svg";
+import ConnectModalBackgroundPattern from "../asset/img/ConnectModalBackgroundPattern.png";
 
 import "./_pera-wallet-modal.scss";
 import "./_pera-wallet-connect-modal.scss";
 
 import React, {useState} from "react";
-import QRCode from "react-qr-code";
 
-import {isLargeScreen} from "../util/screen/screenSizeUtils";
-import {isMobile} from "../util/device/deviceUtils";
 import {
   generatePeraWalletConnectDeepLink,
   getPeraWalletAppMeta
 } from "../util/peraWalletUtils";
+import Accordion from "./component/accordion/Accordion";
+import {useIsSmallScreen} from "../util/screen/useMediaQuery";
+import PeraWalletConnectModalInformationSection from "./section/information/PeraWalletConnectModalInformationSection";
+import PeraWalletConnectModalPendingMessage from "./section/pending-message/PeraWalletConnectModalPendingMessage";
+import {getPeraConnectModalAccordionData} from "./util/peraWalletConnectModalUtils";
 
 interface PeraWalletConnectModalProps {
   uri: string;
   onClose: () => void;
 }
 
 function PeraWalletConnectModal({uri, onClose}: PeraWalletConnectModalProps) {
-  const {logo, name, main_color} = getPeraWalletAppMeta();
-  const [isQRCodeVisible, setQRCodeVisibility] = useState(!isMobile());
+  const {name} = getPeraWalletAppMeta();
   const [isSpinnerVisible, setSpinnerVisibility] = useState(false);
```

Comment:
I don't think this is a good name for this state. I think we should treat as "view" modes
```suggestion
  const [view, setView] = useState("default" as "default" | "launching-app");
```